clap_man/src/lib.rs

```diff
@@ -0,0 +1,105 @@
+#![doc(html_logo_url = "https://raw.githubusercontent.com/clap-rs/clap/master/assets/clap.png")]
+#![doc = include_str!("../README.md")]
+#![warn(missing_docs, trivial_casts, unused_allocation, trivial_numeric_casts)]
+#![forbid(unsafe_code)]
+
+mod man;
+mod render;
+
+use std::io::Write;
+
+/// Man page generator
+#[derive(Debug, Clone)]
+pub struct Man {
+    section: Option<i8>,
+    manual: Option<String>,
+    sections: Vec<(String, Vec<String>)>,
+}
+
+impl Default for Man {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+/// Generate manpage for your application using sane defaults.
+pub fn generate_manpage<'a>(app: &mut clap::App<'a>, buf: &mut dyn Write) {
+    let man = Man::new().section(1).manual("General Commands Manual");
+    man.render(app, buf);
+}
+
+impl Man {
+    /// Create a new builder for man pages.
+    pub fn new() -> Self {
+        Man {
+            section: None,
+            manual: None,
+            sections: Vec::new(),
+        }
+    }
+
+    /// Add section for your man page, the most common being 1 for programs.
+    ///
+    /// 1. Executable programs or shell commands
+    /// 2. System calls (functions provided by the kernel)
+    /// 3. Library calls (functions within program libraries)
+    /// 4. Special files (usually found in /dev)
+    /// 5. File formats and conventions, e.g. /etc/passwd
+    /// 6. Games
+    /// 7. Miscellaneous (including macro packages and conventions), e.g. man(7), groff(7)
+    /// 8. System administration commands (usually only for root)
+    /// 9. Kernel routines [Non standard]
```

Comment:
Should we provide either constants for this or even switch to an enum?

Reply:
I pulled this out to `roff-rs`.